node/test/health.js

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+/* jshint maxparams:5 */
+/*eslint max-params: [2, 5]*/
+
+'use strict';
+
+var path = require('path');
+var fs = require('fs');
+
+var TChannelAsThrift = require('../as/thrift.js');
+var allocCluster = require('./lib/alloc-cluster.js');
+
+var globalThriftText = fs.readFileSync(
+    path.join(__dirname, 'anechoic-chamber.thrift'), 'utf8'
+);
+
+allocCluster.test('thrift works with health check present', {
+    numPeers: 2
+}, function t(cluster, assert) {
+    var tchannelAsThrift = makeTChannelThriftServer(cluster, {
+        okResponse: true
+    });
+
+    var client = cluster.channels[1].subChannels.server;
+
+    tchannelAsThrift.send(client.request({
+        serviceName: 'server',
+        hasNoParent: true
+    }), 'Chamber::echo', null, {
+        value: 10
+    }, function onResponse(err, res) {
+        assert.ifError(err);
```

Comment:
please use the `if (err) return assert.end(err)` pattern. Failing tests should not crash the test runner.